src/app/loadout-builder/generated-sets/GeneratedSetItem.tsx

```diff
@@ -15,9 +15,13 @@ import { generateMixesFromPerks } from '../process';
 /**
  * Figure out which (if any) non-selected perks should be selected to get the chosen stat mix.
  */
-function identifyAltPerkChoicesForChosenStats(item: DimItem, chosenValues: number[]) {
+function identifyAltPerkChoicesForChosenStats(
+  item: DimItem,
+  chosenValues: number[],
+  assumeMasterwork: boolean
+) {
   let altPerks: DimPlug[] = [];
-  generateMixesFromPerks(item, (mix, plugs) => {
+  generateMixesFromPerks(item, assumeMasterwork, (mix, plugs) => {
```

Comment:
I had one thought - I'm not sure this matters. For Armor 2.0, there aren't alternate perks to select - we should probably just bail out of this immediately. However, I thought the +2 bonus to all stats only applied to Armor 2.0. That suggests that this is either not going to be applied (because it's Armor 1.0) or it won't matter (because it's Armor 2.0). Thoughts?

Reply:
Yeah the +2 is only ever applied to armour 2.0, that is handled in process by checking isDestiny2 and the energy attribute. I have a feeling this is getting run every time whether it is needed or not. I will put a guard in around it and see if it improves things at all.

Reply:
I could stop passing it thought as props and just set this to always use false. Then may at least stop some GeneratedSetItem re-renders.

Reply:
Yeah I think the big thing is not so much the re-renders - I just don't think you needed to plumb this all through these components at all, since it doesn't affect anything.

Reply:
As a comparison, on my machine all of the work before the big many-loops section (grouping by stat mix, etc) takes 6ms. The loops take 1700ms. I don't really worry about that first part.

Reply:
Yeah I agree, I will sort something out after work today. I won't try and completely untangle the logic that's going on just yet but I will remove the need to pass it though and additionally do a check around which items need to have alt perks calculated.

Reply:
Done, have a look at that and let me know what you think.